File: backend/internal/model/log.go
/*
 * @Author              : 寂情啊
 * @Date                : 2025-11-14 15:25:08
 * @LastEditors         : 寂情啊
 * @LastEditTime        : 2025-12-03 16:36:04
 * @FilePath            : frp-web-testbackendinternalmodellog.go
 * @Description         : 说明
 * 倾尽绿蚁花尽开，问潭底剑仙安在哉
 */
package model

import (
	"time"
)

type OperationLog struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	UserID        uint      `json:"user_id" gorm:"index"`
	Username      string    `json:"username" gorm:"-"` // 非数据库字段，用于返回用户名
	OperationType string    `json:"operation_type" gorm:"size:50;not null"`
	ResourceType  string    `json:"resource_type" gorm:"size:50;not null"`
	ResourceID    uint      `json:"resource_id"`
	Description   string    `json:"description" gorm:"type:text"`
	IPAddress     string    `json:"ip_address" gorm:"size:50"`
	IPLocation    string    `json:"ip_location" gorm:"size:100"` // IP归属地
	CreatedAt     time.Time `json:"created_at"`
}

File: web/src/api/log.ts
import request from './request';
import type { PaginationResponse } from '../types';

export interface OperationLog {
  id: number;
  user_id: number;
  username: string;
  operation_type: string;
  resource_type: string;
  resource_id: number;
  description: string;
  ip_address: string;
  ip_location: string;
  created_at: string;
}

export const logApi = {
  getLogs: (params: { page: number; page_size: number; operation_type?: string; resource_type?: string }) =>
    request.get<PaginationResponse<OperationLog>>('/logs', { params }),

  createLog: (data: Partial<OperationLog>) => request.post('/logs', data)
};